src/script/interpreter.cpp

```diff
@@ -1536,11 +1538,25 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         if (uses_bip341_taproot && uses_bip143_segwit) break; // No need to scan further if we already need all.
     }
 
-    if (uses_bip143_segwit || uses_bip341_taproot) {
+    if (uses_bip143_segwit || uses_bip341_taproot || uses_bip119_ctv) {
         // Computations shared between both sighash schemes.
         m_prevouts_single_hash = GetPrevoutsSHA256(txTo);
         m_sequences_single_hash = GetSequencesSHA256(txTo);
         m_outputs_single_hash = GetOutputsSHA256(txTo);
+
+        bool skip_scriptSigs = std::find_if(txTo.vin.begin(), txTo.vin.end(),
```

Comment:
This section duplicates large parts of the logic from `GetDefaultCheckTemplateVerifyHash()`. Would it be possible to have this logic in one place?
Also, after making reliance on the cache obligatory in a later commit, it seems that `GetDefaultCheckTemplateVerifyHash` is only used in a test anymore (`ctvhash_tests`), which in my opinion at least deserves a comment.

Reply:
this is a good point. I guess it's a question of if it's better to have the functionality available as a part of the API, or if it can be moved to a test.

Reply:
note that using the cache is mandatory, and using the non-cached version of GetDefaultCheckTemplateVerifyHash from e.g. script is insecure.